test/e2e/container_inspect_test.go

```diff
@@ -82,4 +82,24 @@ var _ = Describe("Podman container inspect", func() {
 		Expect(data[0].HostConfig.VolumesFrom).To(Equal([]string{volsctr}))
 		Expect(data[0].Config.Annotations[define.VolumesFromAnnotation]).To(Equal(volsctr))
 	})
+
+	It("podman inspect hides secrets mounted to env", func() {
+		secretName := "mysecret"
+		secretValue := "mysecretValue"
+
+		session := podmanTest.Podman([]string{"secret", "create", secretName, secretValue})
+
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		name := "testcon"
+		session = podmanTest.Podman([]string{"run", "--secret", fmt.Sprintf("%s,type=env", secretName), "--name", name, CITEST_IMAGE})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		data := podmanTest.InspectContainer(name)
+		Expect(data).To(HaveLen(1))
+		Expect(data[0].Config.Env).To(HaveLen(1))
+		Expect(data[0].Config.Env[0]).To(HavePrefix(secretName+"="), Not(HaveSuffix(secretValue)))
```

Comment:
This isn't doing what you think. The second argument to To() is a description not a a matcher so the `Not(HaveSuffix(...))` part is not checked.
In general it is always best to do exact matches, i.e. `Equal(secretName+"=*******")`